lang/src/org/partiql/lang/typesystem/builtin/types/nonprimitive/DoublePrecisionType.kt

```diff
@@ -0,0 +1,18 @@
+package org.partiql.lang.typesystem.builtin.types.nonprimitive
+
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.typesystem.interfaces.type.BuiltInType
+
+/**
+ * DOUBLE_PRECISION type is actually FLOAT8
+ */
+object DoublePrecisionType : BuiltInType() {
```

Comment:
Why aren't we considering this and other `int` and `float` types as primitive?

Reply:
There is no `DoublePrecision` as one of `ExprValueType`.

Reply:
So basically, I want to classify types into primitive types and non-primitive types. If a sql type corresponds to one `ExprValueType`, it is a primitive type. As far as I can see, this classification is not used anywhere, but I think it helps us to understand differences between types.

Reply:
I still need to think about this more, but for now, if we're going with this categorization we should make it clear in in `kdoc`.